Patch postinst so that the package installs on Debian Jessie.

vulpicastor opened the pull request:
- `debian/debathena-pyhesiodfs.postinst`: Add user pyhesiodfs to group fuse
  only if the local group exists. As of Debian Jessie, the group fuse is no
  longer used. ([Trac: #1523](https://athena10.mit.edu/trac/ticket/1523))

Signed-off-by: Lizhou Sha slz@mit.edu

vasilvv commented:
We've already discussed this on Zephyr.  I suggested that it might be explicitly desirable to check against the local group repository, since the existence of fuse group in Hesiod/YP is not in particular meaningful.  Especially given that Moira actually has a fuse group.

andersk commented on debian/debathena-pyhesiodfs.postinst:
```diff
@@ -25,8 +25,17 @@ set -e
+    # The group fuse no longer exists in jessie.
+    # If the group fuse is needed, it should have been added when the package 
+    # fuse, a dependency, was configured, before this script is run.
+    if grep "^fuse:" /etc/group > /dev/null 2>&1; then
```
`if grep -q "^fuse:" /etc/group; then`

(It’s always better to avoid suppressing more errors than we’re expecting.)

vulpicastor replied:
Does `grep -q` only suppress `stdout` and not `stderr`? If so, then according to the GNU grep man page, `grep "^fuse:" /etc/group > /dev/null` might match that behavior the best.

andersk commented on debian/debathena-pyhesiodfs.postinst:
```diff
@@ -25,8 +25,17 @@ set -e
-        
+    else
+        echo "The group fuse does not exist on your system. Skipping adding pyhesiodfs to it." >&2
+    fi
```
This block (`if`…`fi` and the preceding comments) needs more indentation.

vulpicastor replied:
This script apparently has mixed tab and space indenting... I did not discover it at first as I had set tab length to 4. I'm not sure what style Debathena follows, but I'm just going to turn all the tab space into spaces, as it doesn't affect most lines.

andersk commented:
The note in the GNU grep manpage that you’re looking at dates back to [seventeen years ago](http://git.savannah.gnu.org/cgit/grep.git/commit/?id=588c41f4fc2da3b75f183a26cb1151cca61ff95b). Any modern grep implementation supports `-q`, and as the manpage also notes, it is specified by POSIX. A Debian postinst script need not concern itself with portability to non-GNU grep, let alone to an ancient non-POSIX grep. `-q` is the idiomatic way to use grep as a conditional.

If you customize your tab width to 4 spaces, you are going to confuse yourself and everyone else. A Unix tab is always 8 spaces wide. This does not imply that indentation levels must be 8 spaces wide; it just means that you don’t use tabs for indentation levels that aren’t 8 spaces wide. The Emacs default is to use 4 space indentations but to compress an initial multiple of 8 spaces using tabs. It may not the prettiest thing in the world, depending on who you ask, but it’s very common. If you’d like to avoid tabs in your own projects, that’s entirely reasonable. But in an existing project, it’s not a good idea to “fix” someone else’s tabs to spaces, because that causes a bunch of unnecessary version control churn that makes it harder to use `git blame` effectively.

vulpicastor commented:
OK. Not sure if resetting and `git push -f` is the right solution, but I've gotten rid of the unnecessary commits. I've fixed the problems, and the patch should now be ready for merging.

jdreed commented:
If you squash those into a single commit, I’ll take the patch.

(This is “git rebase -i” and then "push -f" in your branch)

Github will, I think, update the pull request automatically.

On Apr 16, 2015, at 3:00 PM, Lizhou Sha <notifications@github.com<mailto:notifications@github.com>> wrote:

OK. Not sure if resetting and git push -f is the right solution, but I've gotten rid of the unnecessary commits. I've fixed the problems, and the patch should now be ready for merging.

—
Reply to this email directly or view it on GitHubhttps://github.com/mit-athena/pyhesiodfs/pull/3#issuecomment-93815511.

vulpicastor commented:
OK. Thoroughly squashed!

vulpicastor committed:
Patch postinst so that the package installs on Debian Jessie.

- debian/debathena-pyhesiodfs.postinst: Add user pyhesiodfs to group
  fuse only if the local group exists. As of Debian Jessie, the group
  fuse is no longer used. (Trac: #1523)

Signed-off-by: Lizhou Sha <slz@mit.edu>

vulpicastor commented:
I think my newest squash and `git push -f` accidentally got rid of kaduk's comment. I ran `dch -r` and suppressed the informational output in the `else` statement, as suggested.

andersk commented:
(Losing the comment wasn’t your fault. Line comments should go to the diff on the [“Files changed” tab](https://github.com/mit-athena/pyhesiodfs/pull/3/files) on the pull request, not to the diff on individual commits, so that they don’t get lost on force-pushes.)

vulpicastor commented:
(For reference, the comment was [here](https://github.com/vulpicastor/pyhesiodfs/commit/4921c2eabc8165c688bf3b6f76f304c1813cd48c#commitcomment-10789528)).

achernya commented:
Merged as faf196dab6cf2bf81a24efd9e8da1470ecc42680.
